algorithms/src/Kokkos_Random.hpp

```diff
@@ -886,29 +886,27 @@ class Random_XorShift64_Pool {
  public:
   using generator_type = Random_XorShift64<DeviceType>;
 
-  KOKKOS_INLINE_FUNCTION
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  KOKKOS_FUNCTION
   Random_XorShift64_Pool() {
-    num_states_ = 0;
-    padding_    = 0;
+    KOKKOS_IF_ON_DEVICE((
+        /* nonsensical initialization in the name of backward compatibility */
+        num_states_ = 0; padding_ = 0;))
+    KOKKOS_IF_ON_HOST(init(0, execution_space().concurrency());)
   }
+
   Random_XorShift64_Pool(uint64_t seed) {
     num_states_ = 0;
 
     init(seed, execution_space().concurrency());
   }
+#else
+  Random_XorShift64_Pool(uint64_t seed = 0) {
+    num_states_ = 0;
 
-  KOKKOS_INLINE_FUNCTION
-  Random_XorShift64_Pool(const Random_XorShift64_Pool& src)
-      : locks_(src.locks_), state_(src.state_), num_states_(src.num_states_) {}
-
-  KOKKOS_INLINE_FUNCTION
-  Random_XorShift64_Pool operator=(const Random_XorShift64_Pool& src) {
-    locks_      = src.locks_;
-    state_      = src.state_;
-    num_states_ = src.num_states_;
-    padding_    = src.padding_;
-    return *this;
+    init(seed, execution_space().concurrency());
   }
+#endif
 
   void init(uint64_t seed, int num_states) {
```

Comment:
why is init a public method? I think init should be private

Reply:
Actually it makes sense to provide a functionality to reseed the random generator engine.

Reply:
I think if we want a method to reseed it, I personally would make it: 
```
void reseed(uint64_t newSeed)
```
becaue `init` is too general, and reseeding does not need the num_states

Reply:
Since the default-constructor behavior was restored to an "uninitialized" state, I am not proposing any change to that function.